storage/badger/chunkDataPacks.go

```diff
@@ -82,6 +82,24 @@ func (ch *ChunkDataPacks) BatchStore(c *flow.ChunkDataPack, batch storage.BatchS
 	return operation.BatchInsertChunkDataPack(sc)(writeBatch)
 }
 
+// StoreMul stores multiple ChunkDataPacks cs keyed by their ChunkIDs in provided batch.
+// No errors are expected during normal operation, but it may return generic error
+func (ch *ChunkDataPacks) StoreMul(cs []*flow.ChunkDataPack) error {
+	batch := NewBatch(ch.db)
+	for _, c := range cs {
+		err := ch.BatchStore(c, batch)
+		if err != nil {
+			return fmt.Errorf("cannot store chunk data pack: %w", err)
+		}
+	}
+
+	err := batch.Flush()
+	if err != nil {
+		return fmt.Errorf("cannot flush batch: %w", err)
+	}
+	return nil
```

Comment:
why do we need to flush the batch here? `BatchInsertChunkDataPack` takes care of this, right?

Reply:
ah nevermind, I see you need it for the `OnScucceed` callback